pkg/stats/statser_prometheus.go

```diff
@@ -0,0 +1,78 @@
+package stats
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/atlassian/gostatsd"
+)
+
+// NOTE: is using a collector/vec of gauges/counters reasonable? Or does
+// using only a single gauge/counter suffice?
+
+// this depends on the metrics being collected, if they are the same thing
+// but can be partitioned into different types/groups, e.g. http requests partitioned
+// by user age and demographics, then using a collector makes sense. (is this correct?)
+
+// PrometheusStatser is a Statser that monitors gostasd's internal metrics from
+// Prometheus, it is useful when there is a large number of ephemeral hosts.
+type PrometheusStatser struct {
+	flushNotifier
+
+	// collector of gauges that stores the internal gauge metrics of gostatsd
```

Comment:
Each `GaugeVec` and `CounterVec` is a single metric name, they need to be lazily created.

Reply:
Does it make sense to use the `Gauges` and `Counter` structs `gostatsd` already provides or would using the structs provided from the `prometheus` package be better?

Reply:
You'd need to use the ones from the prometheus packages, because they're ultimately added to the prometheus registry.